vars/infra.groovy

```diff
@@ -29,29 +29,57 @@ Object checkout(String repo = null) {
     }
 }
 
+/**
+ * Retrieves Settings file to be used with Maven.
+ * If {@code MAVEN_SETTINGS_FILE_ID} variable is defined, the file will be retrieved from the specified
+ * configuration ID provided by Config File Provider Plugin.
+ * Otherwise it will fallback to a standard Jenkins infra resolution logic.
+ * @param settingsXml Absolute path to the destination settings file
+ * @return {@code true} if the file has been defined
+ */
+boolean retrieveMavenSettingsFile(String settingsXml) {
+    if (env.MAVEN_SETTINGS_FILE_ID != null) {
+        configFileProvider([configFile(fileId: env.MAVEN_SETTINGS_FILE_ID, variable: 'mvnSettingsFile')]) {
+            if (isUnix()) {
+                sh "cp ${mvnSettingsFile} ${settingsXml}"
+            } else {
+                bat "copy ${mvnSettingsFile} ${settingsXml}"
+            }
+            return true
+        }
+    } else if (jdk.toInteger() > 7 && isRunningOnJenkinsInfra()) {
```

Comment:
Is failing me with jdk is not defined

Reply:
is this still a problem for you @raul-arabaolaza

Reply:
@rtyler No, working properly for me now (via replay) in my testing environment